media/js/firefox/new/experiment-firefox-new-waitface.js

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+/* Experiment: https://bugzilla.mozilla.org/show_bug.cgi?id=1434764
+ * URL: https://www.mozilla.org/en-US/firefox/new/?xv=waitface
+ * Audience: Desktop, Non-Fx, en-US with utm_medium=display&utm_source=doubleclick    
+ * Traffic: 100% audience, 50/50 split*/
+
+(function() {
+    'use strict';
+
+    var isDesktop = function() {
+        return ! /android|ios/.test(site.platform);
+    };
+
+    var notFirefox = function () {
+        return ! /\sFirefox/.test(navigator.userAgent);
+    };
+
+    var paramsMatch = function() {
+        var params = {
+            'utm_medium': 'display', 
+            'utm_source': 'doubleclick'
+        };
+        for(var key in params) {
+            if (! (new RegExp('[\&\?]' + key + '=' + params[key])).test(location.search)) {
+                return false;
+            }
+        }
+        return true;
+    };
+
+    if (isDesktop() && notFirefox() && paramsMatch()) {
+        var cop = new Mozilla.TrafficCop({
+            id: 'experiment_firefox_new_waitface',
+            variations: {
+                novariation: 50,
```

Comment:
This experiment is mixing both a redirect and a callback variation, which is something I'm not sure we've done before. The [docs](https://github.com/mozilla/trafficcop/blob/master/documentation.md) don't specifically say we should/should not do this, but one thing I'm not sure about is how we would track the double-control group here without a unique URL. Thoughts on this @jpetto?

Reply:
Also note that in the past we've often supplied a control variation to account for the redirect time. E.g., you may want to add a `'v=control': 50` that will simply redirect to control template.